source/models/row.py

```diff
@@ -0,0 +1,60 @@
+import os, json, uuid
+from commands_functions.schema_keys import Keys
+from outputs.exceptions import *
+from models.index import Index
+
+class Row:
+    def __init__(self, table, value = None, primary_key = None):
+        self.table = table
+        if value is not None:
+            self.__init_by_value(value)
+        elif primary_key is not None:
+            self.__init_by_primary_key()
+        else:
+            raise WrongInput(message = "the value and PK of the row is null")
+        
+    def __init_by_value(self, value):
+        self.value = value
+        self.primary_key = value[self.table.table_metadata.primary_key]
```

Comment:
In this case, primary key can be null. I would suggest set primary_key with random variable if it is null and update data[primary_key]

Reply:
I did so in the set command. so I suppose that the entered value is containing the PK if it is a given one or a random one,
so why it is preferred to be here?

Reply:
ok done